rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -179,22 +183,21 @@ class ReplayRSpace[F[_]: Sync, C, P, A, K](
       channels =>
         store
           .getContinuations(channels)
-          .map(_.zipWithIndex.filter {
+          .map(_.iterator.zipWithIndex.filter {
```

Comment:
Why is this more performance than just `zipWithIndex`? Whatever the answer is, i think it'd be a good idea to leave a comment about why you're using `iterator` + `toSeq` instead of `zipWithIndex`.

Reply:
When you do `seq.zipWithIndex.filter` you're allocating two collections. One that is the result of `seq.zipWithIndex` and the other that is the first one filtered. With `.iterator` the intermediate collection is eliminated, and allocation is constrained to only one (possibly much smaller) collection

Reply:
So, to be really descriptive, if you have like 100 events returned from the store, what is happening without the iterator is:
```
temp1 = new Builder
for every element at i { temp1 += (element, i) }
afterZipWithIndex = temp1.makeSeq
temp2 = new Builder
for every element of afterZipWithIndex { if (p) temp2 += element }
res = temp2.makeSeq
```

wheras with the iterator:
```
orig = elements.iterator
temp1 = new Iterator { override next = (orig.next, i) }
temp2 = new Iterator { override next = if p(temp1.next) .. }
temp2.toSeq
```